scripts/amex_browseros_bridge.py

```
@@ -0,0 +1,543 @@
+
+def first_ref_for_role(snapshot: str, role: str) -> str:
+    match = re.search(rf"- {re.escape(role)} [^\n]*\[ref=(e\d+)\]", snapshot)
+    if match is None:
+        raise RuntimeError(f"Amex page changed: no {role} is available")
+    return match.group(1)
+
+
+def amex_url(snapshot: str) -> str:
+    match = re.search(r"origin=(https://[^\s\]]+)", snapshot)
+    if match is None:
+        raise RuntimeError("BrowserOS did not return the active page URL")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)